src/compas/_iotools.py

```diff
@@ -42,8 +42,9 @@ def open_file(file_or_filename, mode='r'):
     if not hasattr(file, 'read'):
         # If it's a URL, open and read its response into an in-memory stream
         if hasattr(file, 'startswith') and file.startswith('http'):
-            if mode != 'r':
-                raise ValueError('URLs can only be opened in read mode')
+            # support all read modes (r, r+, rb, etc)
+            if not mode.startswith('r'):
```

Comment:
mode `'r+'` starts with 'r' but also is writable.  should that be allowed?

Reply:
I would say so. Especially for the cases in which `r+` is used, but never upgraded for write (one could claim then `r+` is being used incorrectly but it's not on us to judge that 😉)